src/cephadm/cephadm

```diff
@@ -7080,6 +7057,141 @@ def command_check_host(ctx: CephadmContext) -> None:
 ##################################
 
 
+def get_ssh_vars(ssh_user: str) -> Tuple[int, int, str]:
+    try:
+        s_pwd = pwd.getpwnam(ssh_user)
+    except KeyError:
+        raise Error('Cannot find uid/gid for ssh-user: %s' % (ssh_user))
+
+    ssh_uid = s_pwd.pw_uid
+    ssh_gid = s_pwd.pw_gid
+    ssh_dir = os.path.join(s_pwd.pw_dir, '.ssh')
+    return ssh_uid, ssh_gid, ssh_dir
+
+
+def authorize_ssh_key(ssh_pub_key: str, ssh_user: str) -> bool:
+    """Authorize the public key for the provided ssh user"""
+
+    def key_in_file(path: str, key: str) -> bool:
+        if not os.path.exists(path):
+            return False
+        with open(path) as f:
+            lines = f.readlines()
+            for line in lines:
+                if line.strip('\n') == key.strip('\n'):
+                    return True
+        return False
+
+    logger.info(f'Adding key to {ssh_user}@localhost authorized_keys...')
+    if ssh_pub_key is None or ssh_pub_key.isspace():
+        raise Error('Trying to authorize an empty ssh key')
+
+    ssh_uid, ssh_gid, ssh_dir = get_ssh_vars(ssh_user)
+    if not os.path.exists(ssh_dir):
+        makedirs(ssh_dir, ssh_uid, ssh_gid, 0o700)
+
+    auth_keys_file = '%s/authorized_keys' % ssh_dir
+    if key_in_file(auth_keys_file, ssh_pub_key):
+        logger.info(f'key already in {ssh_user}@localhost authorized_keys...')
+        return False
+
+    add_newline = False
+    if os.path.exists(auth_keys_file):
+        with open(auth_keys_file, 'r') as f:
+            f.seek(0, os.SEEK_END)
+            if f.tell() > 0:
+                f.seek(f.tell() - 1, os.SEEK_SET)  # go to last char
+                if f.read() != '\n':
+                    add_newline = True
+
+    with open(auth_keys_file, 'a') as f:
+        os.fchown(f.fileno(), ssh_uid, ssh_gid)  # just in case we created it
+        os.fchmod(f.fileno(), 0o600)  # just in case we created it
+        if add_newline:
+            f.write('\n')
+        f.write(ssh_pub_key + '\n')
+
+    return True
+
+
+def revoke_ssh_key(key: str, ssh_user: str) -> None:
+    """Revoke the public key authorization for the ssh user"""
+    ssh_uid, ssh_gid, ssh_dir = get_ssh_vars(ssh_user)
+    auth_keys_file = '%s/authorized_keys' % ssh_dir
+    deleted = False
+    if os.path.exists(auth_keys_file):
+        with open(auth_keys_file, 'r') as f:
+            lines = f.readlines()
+        _, filename = tempfile.mkstemp()
+        with open(filename, 'w') as f:
+            os.fchown(f.fileno(), ssh_uid, ssh_gid)
+            os.fchmod(f.fileno(), 0o600)  # secure access to the keys file
+            for line in lines:
+                if line.strip('\n') == key.strip('\n'):
+                    deleted = True
+                else:
+                    f.write(line)
+
+    if deleted:
+        shutil.move(filename, auth_keys_file)
+    else:
+        logger.warning('Cannot find the ssh key to be deleted')
```

Comment:
@rkachach  Did some test around this PR. It mostly works, and seemed to be able to correctly detect if the user has passwordless sudo access with or without providing custom keys. However, there was one bug around  the writing to and cleaning up the authorized keys file. For one, I tested without providing explicit pub/priv keys (and with an empty authorixed_keys file for my user "test"). Each time it check the ssh connection and failed it added an additional newline to the authorized keys file.
```
[test@vm-00 ~]$ cat /home/test/.ssh/authorized_keys 




[test@vm-00 ~]$ sudo /root/cephadm --image quay.io/adk3798/ceph:testing bootstrap --mon-ip 192.168.122.249 --ssh-user test
Verifying ssh connectivity ...
Adding key to test@localhost authorized_keys...
Non-zero exit code 1 from ssh -o StrictHostKeyChecking=no -i /tmp/ssh_key_c3043554-ccad-11ec-b438-525400049d84 -o PasswordAuthentication=no test@vm-00 sudo echo
ssh: stderr sudo: a terminal is required to read the password; either use the -S option to read from standard input or configure an askpass helper
ERROR: 
** Please verify your user's ssh configuration and make sure:
- User test must have passwordless sudo access


[test@vm-00 ~]$ sudo /root/cephadm --image quay.io/adk3798/ceph:testing bootstrap --mon-ip 192.168.122.249 --ssh-user test
Verifying ssh connectivity ...
Adding key to test@localhost authorized_keys...
Non-zero exit code 1 from ssh -o StrictHostKeyChecking=no -i /tmp/ssh_key_c5800894-ccad-11ec-98f1-525400049d84 -o PasswordAuthentication=no test@vm-00 sudo echo
ssh: stderr sudo: a terminal is required to read the password; either use the -S option to read from standard input or configure an askpass helper
ERROR: 
** Please verify your user's ssh configuration and make sure:
- User test must have passwordless sudo access


[test@vm-00 ~]$ sudo /root/cephadm --image quay.io/adk3798/ceph:testing bootstrap --mon-ip 192.168.122.249 --ssh-user test
Verifying ssh connectivity ...
Adding key to test@localhost authorized_keys...
Non-zero exit code 1 from ssh -o StrictHostKeyChecking=no -i /tmp/ssh_key_c7bdd294-ccad-11ec-8d28-525400049d84 -o PasswordAuthentication=no test@vm-00 sudo echo
ssh: stderr sudo: a terminal is required to read the password; either use the -S option to read from standard input or configure an askpass helper
ERROR: 
** Please verify your user's ssh configuration and make sure:
- User test must have passwordless sudo access


[test@vm-00 ~]$ cat /home/test/.ssh/authorized_keys 







[test@vm-00 ~]$
```

Then, I tested with a pub/priv key pair with the pub key present in the authorized keys file. This caused a bigger issue in that the key was left written multiple times in the authorized keys file.

```
[test@vm-00 ~]$ cat /home/test/.ssh/authorized_keys 












[test@vm-00 ~]$ cat /home/test/.ssh/id_rsa.pub | cat >> /home/test/.ssh/id_rsa.pub 
[test@vm-00 ~]$ cat /home/test/.ssh/id_rsa.pub 
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCojk0wYD7xTAExhl/lgD75xOc9jB1mU/FgPa/nNFWq+kTcZicvV0KmE/TaSdZMxyaTxs5zD7TG5Q/3CciXAtkImb19Lcdjui7VQFp5PrQ/vz+f2TlSec0iYDQ5npokxWlE/68i2OjehJ/Q2aO++ftzEluW8BxrX+g/tMlCCfkrTFbXn/I2p+5Nk7heV9dWIq22RplF3Gkvu0X0mwVHNnzDPOB9XOdB1mNlRkpOsm45CswRuw4KOkNKICF6WlN9Wth/avXgQukCy4M+/1Si4NgW1Dli80xUSHz7xw0f1lE0RYbtjxsRgP3x7HTNgzOXNiQw10iyrymSLCmMNTPBiE/r3EJMugOa4T1sC2bWTPELc713sqk/fIXSyEsXM2Ckg+9G7pchEzCW7DDC5M6XOgdh/yrnQ1znicGdnKW1XSEo9A4m0Cyv2BJeBTMI/9UCCS7n4PzlrbNPFglr+QPCcPgZdvn8xmdM/Wnmb670tR8H0MtqmoKdC2vKUI5FECRv52s= test@vm-00.test2.com
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCojk0wYD7xTAExhl/lgD75xOc9jB1mU/FgPa/nNFWq+kTcZicvV0KmE/TaSdZMxyaTxs5zD7TG5Q/3CciXAtkImb19Lcdjui7VQFp5PrQ/vz+f2TlSec0iYDQ5npokxWlE/68i2OjehJ/Q2aO++ftzEluW8BxrX+g/tMlCCfkrTFbXn/I2p+5Nk7heV9dWIq22RplF3Gkvu0X0mwVHNnzDPOB9XOdB1mNlRkpOsm45CswRuw4KOkNKICF6WlN9Wth/avXgQukCy4M+/1Si4NgW1Dli80xUSHz7xw0f1lE0RYbtjxsRgP3x7HTNgzOXNiQw10iyrymSLCmMNTPBiE/r3EJMugOa4T1sC2bWTPELc713sqk/fIXSyEsXM2Ckg+9G7pchEzCW7DDC5M6XOgdh/yrnQ1znicGdnKW1XSEo9A4m0Cyv2BJeBTMI/9UCCS7n4PzlrbNPFglr+QPCcPgZdvn8xmdM/Wnmb670tR8H0MtqmoKdC2vKUI5FECRv52s= test@vm-00.test2.com
[test@vm-00 ~]$ 
[test@vm-00 ~]$ sudo /root/cephadm --image quay.io/adk3798/ceph:testing bootstrap --mon-ip 192.168.122.249 --ssh-user test --ssh-private-key /home/test/.ssh/id_rsa --ssh-public-key /home/test/.ssh/id_rsa.pub
Verifying ssh connectivity ...
Adding key to test@localhost authorized_keys...
Non-zero exit code 1 from ssh -o StrictHostKeyChecking=no -i /home/test/.ssh/id_rsa -o PasswordAuthentication=no test@vm-00.test2.com sudo echo
ssh: stderr sudo: a terminal is required to read the password; either use the -S option to read from standard input or configure an askpass helper
Cannot find the ssh key to be deleted
ERROR: 
** Please verify your user's ssh configuration and make sure:
- User test must have passwordless sudo access
- The public key file configured by --ssh-public-key is valid
- The private key file configured by --ssh-private-key is valid


[test@vm-00 ~]$ sudo /root/cephadm --image quay.io/adk3798/ceph:testing bootstrap --mon-ip 192.168.122.249 --ssh-user test --ssh-private-key /home/test/.ssh/id_rsa --ssh-public-key /home/test/.ssh/id_rsa.pub
Verifying ssh connectivity ...
Adding key to test@localhost authorized_keys...
Non-zero exit code 1 from ssh -o StrictHostKeyChecking=no -i /home/test/.ssh/id_rsa -o PasswordAuthentication=no test@vm-00.test2.com sudo echo
ssh: stderr sudo: a terminal is required to read the password; either use the -S option to read from standard input or configure an askpass helper
Cannot find the ssh key to be deleted
ERROR: 
** Please verify your user's ssh configuration and make sure:
- User test must have passwordless sudo access
- The public key file configured by --ssh-public-key is valid
- The private key file configured by --ssh-private-key is valid


[test@vm-00 ~]$ cat /home/test/.ssh/authorized_keys 












ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCojk0wYD7xTAExhl/lgD75xOc9jB1mU/FgPa/nNFWq+kTcZicvV0KmE/TaSdZMxyaTxs5zD7TG5Q/3CciXAtkImb19Lcdjui7VQFp5PrQ/vz+f2TlSec0iYDQ5npokxWlE/68i2OjehJ/Q2aO++ftzEluW8BxrX+g/tMlCCfkrTFbXn/I2p+5Nk7heV9dWIq22RplF3Gkvu0X0mwVHNnzDPOB9XOdB1mNlRkpOsm45CswRuw4KOkNKICF6WlN9Wth/avXgQukCy4M+/1Si4NgW1Dli80xUSHz7xw0f1lE0RYbtjxsRgP3x7HTNgzOXNiQw10iyrymSLCmMNTPBiE/r3EJMugOa4T1sC2bWTPELc713sqk/fIXSyEsXM2Ckg+9G7pchEzCW7DDC5M6XOgdh/yrnQ1znicGdnKW1XSEo9A4m0Cyv2BJeBTMI/9UCCS7n4PzlrbNPFglr+QPCcPgZdvn8xmdM/Wnmb670tR8H0MtqmoKdC2vKUI5FECRv52s= test@vm-00.test2.com
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCojk0wYD7xTAExhl/lgD75xOc9jB1mU/FgPa/nNFWq+kTcZicvV0KmE/TaSdZMxyaTxs5zD7TG5Q/3CciXAtkImb19Lcdjui7VQFp5PrQ/vz+f2TlSec0iYDQ5npokxWlE/68i2OjehJ/Q2aO++ftzEluW8BxrX+g/tMlCCfkrTFbXn/I2p+5Nk7heV9dWIq22RplF3Gkvu0X0mwVHNnzDPOB9XOdB1mNlRkpOsm45CswRuw4KOkNKICF6WlN9Wth/avXgQukCy4M+/1Si4NgW1Dli80xUSHz7xw0f1lE0RYbtjxsRgP3x7HTNgzOXNiQw10iyrymSLCmMNTPBiE/r3EJMugOa4T1sC2bWTPELc713sqk/fIXSyEsXM2Ckg+9G7pchEzCW7DDC5M6XOgdh/yrnQ1znicGdnKW1XSEo9A4m0Cyv2BJeBTMI/9UCCS7n4PzlrbNPFglr+QPCcPgZdvn8xmdM/Wnmb670tR8H0MtqmoKdC2vKUI5FECRv52s= test@vm-00.test2.com

ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCojk0wYD7xTAExhl/lgD75xOc9jB1mU/FgPa/nNFWq+kTcZicvV0KmE/TaSdZMxyaTxs5zD7TG5Q/3CciXAtkImb19Lcdjui7VQFp5PrQ/vz+f2TlSec0iYDQ5npokxWlE/68i2OjehJ/Q2aO++ftzEluW8BxrX+g/tMlCCfkrTFbXn/I2p+5Nk7heV9dWIq22RplF3Gkvu0X0mwVHNnzDPOB9XOdB1mNlRkpOsm45CswRuw4KOkNKICF6WlN9Wth/avXgQukCy4M+/1Si4NgW1Dli80xUSHz7xw0f1lE0RYbtjxsRgP3x7HTNgzOXNiQw10iyrymSLCmMNTPBiE/r3EJMugOa4T1sC2bWTPELc713sqk/fIXSyEsXM2Ckg+9G7pchEzCW7DDC5M6XOgdh/yrnQ1znicGdnKW1XSEo9A4m0Cyv2BJeBTMI/9UCCS7n4PzlrbNPFglr+QPCcPgZdvn8xmdM/Wnmb670tR8H0MtqmoKdC2vKUI5FECRv52s= test@vm-00.test2.com
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCojk0wYD7xTAExhl/lgD75xOc9jB1mU/FgPa/nNFWq+kTcZicvV0KmE/TaSdZMxyaTxs5zD7TG5Q/3CciXAtkImb19Lcdjui7VQFp5PrQ/vz+f2TlSec0iYDQ5npokxWlE/68i2OjehJ/Q2aO++ftzEluW8BxrX+g/tMlCCfkrTFbXn/I2p+5Nk7heV9dWIq22RplF3Gkvu0X0mwVHNnzDPOB9XOdB1mNlRkpOsm45CswRuw4KOkNKICF6WlN9Wth/avXgQukCy4M+/1Si4NgW1Dli80xUSHz7xw0f1lE0RYbtjxsRgP3x7HTNgzOXNiQw10iyrymSLCmMNTPBiE/r3EJMugOa4T1sC2bWTPELc713sqk/fIXSyEsXM2Ckg+9G7pchEzCW7DDC5M6XOgdh/yrnQ1znicGdnKW1XSEo9A4m0Cyv2BJeBTMI/9UCCS7n4PzlrbNPFglr+QPCcPgZdvn8xmdM/Wnmb670tR8H0MtqmoKdC2vKUI5FECRv52s= test@vm-00.test2.com

[test@vm-00 ~]$ sudo /root/cephadm --image quay.io/adk3798/ceph:testing bootstrap --mon-ip 192.168.122.249 --ssh-user test --ssh-private-key /home/test/.ssh/id_rsa --ssh-public-key /home/test/.ssh/id_rsa.pub
Verifying ssh connectivity ...
Adding key to test@localhost authorized_keys...
Non-zero exit code 1 from ssh -o StrictHostKeyChecking=no -i /home/test/.ssh/id_rsa -o PasswordAuthentication=no test@vm-00.test2.com sudo echo
ssh: stderr sudo: a terminal is required to read the password; either use the -S option to read from standard input or configure an askpass helper
Cannot find the ssh key to be deleted
ERROR: 
** Please verify your user's ssh configuration and make sure:
- User test must have passwordless sudo access
- The public key file configured by --ssh-public-key is valid
- The private key file configured by --ssh-private-key is valid


[test@vm-00 ~]$ sudo /root/cephadm --image quay.io/adk3798/ceph:testing bootstrap --mon-ip 192.168.122.249 --ssh-user test --ssh-private-key /home/test/.ssh/id_rsa --ssh-public-key /home/test/.ssh/id_rsa.pub
Verifying ssh connectivity ...
Adding key to test@localhost authorized_keys...
Non-zero exit code 1 from ssh -o StrictHostKeyChecking=no -i /home/test/.ssh/id_rsa -o PasswordAuthentication=no test@vm-00.test2.com sudo echo
ssh: stderr sudo: a terminal is required to read the password; either use the -S option to read from standard input or configure an askpass helper
Cannot find the ssh key to be deleted
ERROR: 
** Please verify your user's ssh configuration and make sure:
- User test must have passwordless sudo access
- The public key file configured by --ssh-public-key is valid
- The private key file configured by --ssh-private-key is valid


[test@vm-00 ~]$ cat /home/test/.ssh/authorized_keys 












ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCojk0wYD7xTAExhl/lgD75xOc9jB1mU/FgPa/nNFWq+kTcZicvV0KmE/TaSdZMxyaTxs5zD7TG5Q/3CciXAtkImb19Lcdjui7VQFp5PrQ/vz+f2TlSec0iYDQ5npokxWlE/68i2OjehJ/Q2aO++ftzEluW8BxrX+g/tMlCCfkrTFbXn/I2p+5Nk7heV9dWIq22RplF3Gkvu0X0mwVHNnzDPOB9XOdB1mNlRkpOsm45CswRuw4KOkNKICF6WlN9Wth/avXgQukCy4M+/1Si4NgW1Dli80xUSHz7xw0f1lE0RYbtjxsRgP3x7HTNgzOXNiQw10iyrymSLCmMNTPBiE/r3EJMugOa4T1sC2bWTPELc713sqk/fIXSyEsXM2Ckg+9G7pchEzCW7DDC5M6XOgdh/yrnQ1znicGdnKW1XSEo9A4m0Cyv2BJeBTMI/9UCCS7n4PzlrbNPFglr+QPCcPgZdvn8xmdM/Wnmb670tR8H0MtqmoKdC2vKUI5FECRv52s= test@vm-00.test2.com
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCojk0wYD7xTAExhl/lgD75xOc9jB1mU/FgPa/nNFWq+kTcZicvV0KmE/TaSdZMxyaTxs5zD7TG5Q/3CciXAtkImb19Lcdjui7VQFp5PrQ/vz+f2TlSec0iYDQ5npokxWlE/68i2OjehJ/Q2aO++ftzEluW8BxrX+g/tMlCCfkrTFbXn/I2p+5Nk7heV9dWIq22RplF3Gkvu0X0mwVHNnzDPOB9XOdB1mNlRkpOsm45CswRuw4KOkNKICF6WlN9Wth/avXgQukCy4M+/1Si4NgW1Dli80xUSHz7xw0f1lE0RYbtjxsRgP3x7HTNgzOXNiQw10iyrymSLCmMNTPBiE/r3EJMugOa4T1sC2bWTPELc713sqk/fIXSyEsXM2Ckg+9G7pchEzCW7DDC5M6XOgdh/yrnQ1znicGdnKW1XSEo9A4m0Cyv2BJeBTMI/9UCCS7n4PzlrbNPFglr+QPCcPgZdvn8xmdM/Wnmb670tR8H0MtqmoKdC2vKUI5FECRv52s= test@vm-00.test2.com

ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCojk0wYD7xTAExhl/lgD75xOc9jB1mU/FgPa/nNFWq+kTcZicvV0KmE/TaSdZMxyaTxs5zD7TG5Q/3CciXAtkImb19Lcdjui7VQFp5PrQ/vz+f2TlSec0iYDQ5npokxWlE/68i2OjehJ/Q2aO++ftzEluW8BxrX+g/tMlCCfkrTFbXn/I2p+5Nk7heV9dWIq22RplF3Gkvu0X0mwVHNnzDPOB9XOdB1mNlRkpOsm45CswRuw4KOkNKICF6WlN9Wth/avXgQukCy4M+/1Si4NgW1Dli80xUSHz7xw0f1lE0RYbtjxsRgP3x7HTNgzOXNiQw10iyrymSLCmMNTPBiE/r3EJMugOa4T1sC2bWTPELc713sqk/fIXSyEsXM2Ckg+9G7pchEzCW7DDC5M6XOgdh/yrnQ1znicGdnKW1XSEo9A4m0Cyv2BJeBTMI/9UCCS7n4PzlrbNPFglr+QPCcPgZdvn8xmdM/Wnmb670tR8H0MtqmoKdC2vKUI5FECRv52s= test@vm-00.test2.com
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCojk0wYD7xTAExhl/lgD75xOc9jB1mU/FgPa/nNFWq+kTcZicvV0KmE/TaSdZMxyaTxs5zD7TG5Q/3CciXAtkImb19Lcdjui7VQFp5PrQ/vz+f2TlSec0iYDQ5npokxWlE/68i2OjehJ/Q2aO++ftzEluW8BxrX+g/tMlCCfkrTFbXn/I2p+5Nk7heV9dWIq22RplF3Gkvu0X0mwVHNnzDPOB9XOdB1mNlRkpOsm45CswRuw4KOkNKICF6WlN9Wth/avXgQukCy4M+/1Si4NgW1Dli80xUSHz7xw0f1lE0RYbtjxsRgP3x7HTNgzOXNiQw10iyrymSLCmMNTPBiE/r3EJMugOa4T1sC2bWTPELc713sqk/fIXSyEsXM2Ckg+9G7pchEzCW7DDC5M6XOgdh/yrnQ1znicGdnKW1XSEo9A4m0Cyv2BJeBTMI/9UCCS7n4PzlrbNPFglr+QPCcPgZdvn8xmdM/Wnmb670tR8H0MtqmoKdC2vKUI5FECRv52s= test@vm-00.test2.com

ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCojk0wYD7xTAExhl/lgD75xOc9jB1mU/FgPa/nNFWq+kTcZicvV0KmE/TaSdZMxyaTxs5zD7TG5Q/3CciXAtkImb19Lcdjui7VQFp5PrQ/vz+f2TlSec0iYDQ5npokxWlE/68i2OjehJ/Q2aO++ftzEluW8BxrX+g/tMlCCfkrTFbXn/I2p+5Nk7heV9dWIq22RplF3Gkvu0X0mwVHNnzDPOB9XOdB1mNlRkpOsm45CswRuw4KOkNKICF6WlN9Wth/avXgQukCy4M+/1Si4NgW1Dli80xUSHz7xw0f1lE0RYbtjxsRgP3x7HTNgzOXNiQw10iyrymSLCmMNTPBiE/r3EJMugOa4T1sC2bWTPELc713sqk/fIXSyEsXM2Ckg+9G7pchEzCW7DDC5M6XOgdh/yrnQ1znicGdnKW1XSEo9A4m0Cyv2BJeBTMI/9UCCS7n4PzlrbNPFglr+QPCcPgZdvn8xmdM/Wnmb670tR8H0MtqmoKdC2vKUI5FECRv52s= test@vm-00.test2.com
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCojk0wYD7xTAExhl/lgD75xOc9jB1mU/FgPa/nNFWq+kTcZicvV0KmE/TaSdZMxyaTxs5zD7TG5Q/3CciXAtkImb19Lcdjui7VQFp5PrQ/vz+f2TlSec0iYDQ5npokxWlE/68i2OjehJ/Q2aO++ftzEluW8BxrX+g/tMlCCfkrTFbXn/I2p+5Nk7heV9dWIq22RplF3Gkvu0X0mwVHNnzDPOB9XOdB1mNlRkpOsm45CswRuw4KOkNKICF6WlN9Wth/avXgQukCy4M+/1Si4NgW1Dli80xUSHz7xw0f1lE0RYbtjxsRgP3x7HTNgzOXNiQw10iyrymSLCmMNTPBiE/r3EJMugOa4T1sC2bWTPELc713sqk/fIXSyEsXM2Ckg+9G7pchEzCW7DDC5M6XOgdh/yrnQ1znicGdnKW1XSEo9A4m0Cyv2BJeBTMI/9UCCS7n4PzlrbNPFglr+QPCcPgZdvn8xmdM/Wnmb670tR8H0MtqmoKdC2vKUI5FECRv52s= test@vm-00.test2.com

ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCojk0wYD7xTAExhl/lgD75xOc9jB1mU/FgPa/nNFWq+kTcZicvV0KmE/TaSdZMxyaTxs5zD7TG5Q/3CciXAtkImb19Lcdjui7VQFp5PrQ/vz+f2TlSec0iYDQ5npokxWlE/68i2OjehJ/Q2aO++ftzEluW8BxrX+g/tMlCCfkrTFbXn/I2p+5Nk7heV9dWIq22RplF3Gkvu0X0mwVHNnzDPOB9XOdB1mNlRkpOsm45CswRuw4KOkNKICF6WlN9Wth/avXgQukCy4M+/1Si4NgW1Dli80xUSHz7xw0f1lE0RYbtjxsRgP3x7HTNgzOXNiQw10iyrymSLCmMNTPBiE/r3EJMugOa4T1sC2bWTPELc713sqk/fIXSyEsXM2Ckg+9G7pchEzCW7DDC5M6XOgdh/yrnQ1znicGdnKW1XSEo9A4m0Cyv2BJeBTMI/9UCCS7n4PzlrbNPFglr+QPCcPgZdvn8xmdM/Wnmb670tR8H0MtqmoKdC2vKUI5FECRv52s= test@vm-00.test2.com
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCojk0wYD7xTAExhl/lgD75xOc9jB1mU/FgPa/nNFWq+kTcZicvV0KmE/TaSdZMxyaTxs5zD7TG5Q/3CciXAtkImb19Lcdjui7VQFp5PrQ/vz+f2TlSec0iYDQ5npokxWlE/68i2OjehJ/Q2aO++ftzEluW8BxrX+g/tMlCCfkrTFbXn/I2p+5Nk7heV9dWIq22RplF3Gkvu0X0mwVHNnzDPOB9XOdB1mNlRkpOsm45CswRuw4KOkNKICF6WlN9Wth/avXgQukCy4M+/1Si4NgW1Dli80xUSHz7xw0f1lE0RYbtjxsRgP3x7HTNgzOXNiQw10iyrymSLCmMNTPBiE/r3EJMugOa4T1sC2bWTPELc713sqk/fIXSyEsXM2Ckg+9G7pchEzCW7DDC5M6XOgdh/yrnQ1znicGdnKW1XSEo9A4m0Cyv2BJeBTMI/9UCCS7n4PzlrbNPFglr+QPCcPgZdvn8xmdM/Wnmb670tR8H0MtqmoKdC2vKUI5FECRv52s= test@vm-00.test2.com

[test@vm-00 ~]$
```
I only copied the key into the file myself once, but by the end we had several copies present. I additionally tested not providing the keys to bootstrap but still with the keys present in the authorized_keys files  and it seemed to return to just adding newlines each time.

If we can fix this I think everything else works here